Create init container for Cilium

joestringer opened the pull request:
Refactor the init container logic into a shell script and create a new container for it.

I've pushed an initial tag version for the container based upon the latest master logic, and manually tested this by deploying in a minikube.

Fixes: #5839

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5857)
<!-- Reviewable:end -->

joestringer committed:
contrib: Create new cilium-init container for k8s

This container will improve the ability to review changes to the init
container by factoring the contents of the script out into a separate
file which is much easier to understand than one giant line inside of a
DaemonSet YAML. It is not yet used, the following commit will be the
first consumer.

Signed-off-by: Joe Stringer <joe@covalent.io>

joestringer committed:
examples: Use new cilium-init container from DS YAMLs

Signed-off-by: Joe Stringer <joe@covalent.io>

aanm commented on Makefile:
```diff
@@ -139,6 +139,9 @@ docker-image: clean GIT_VERSION envoy/SOURCE_VERSION
 	$(QUIET)echo "docker push cilium/cilium:$(DOCKER_IMAGE_TAG)"
 
+docker-image-init:
+	$(QUIET)cd contrib/packaging/docker && docker build -t "cilium/cilium-init:$(UTC_DATE)" -f Dockerfile.init .
```
Please create a automatic build repository for `cilium/cilium-init`

nebril replied:
Will `Dockerfile.init` be read by automated docker hub builds? I am not sure, but it seems like automated build settings only accept a directory which should contain `Dockerfile`. If that's the case, this dockerfile should be moved to a separate dir.

joestringer replied:
@nebril You can specify which dockerfile that dockerhub uses :)

joestringer committed:
init-container: Remove pidfile handling

Previously, we needed to remove pidfiles in the k8s init container,
however since commit 71a863422a43 ("daemon: Clean up k8s health EP
pidfile on startup"), this logic has been shifted into the daemon.
Since this logic is no longer needed, we can remove it from the init
container.

Signed-off-by: Joe Stringer <joe@covalent.io>
